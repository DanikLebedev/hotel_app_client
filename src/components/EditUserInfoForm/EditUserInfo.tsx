import { Container } from 'react-bootstrap';
import React, { useContext, useEffect, useState } from 'react';
import { Data } from '../../interfaces/clientInterfaces';
import toaster from 'toasted-notes';
import { Customer } from '../../interfaces/clientInterfaces';
import { CustomerService } from '../../APIServices/customerService';
import { ClientContext } from '../../context/client.context';
import { useForm } from 'react-hook-form';
import { ErrorMessage } from '../ErrorsComponents/ErrorMessage';
import { handleClickOutside } from '../../sharedMethods/outsideClick';
import { Close} from '@material-ui/icons';


type InputEvent = React.ChangeEvent<HTMLInputElement>;

interface EditUserInfoForm {
    closeModal: () => void;
    show: boolean;
    editProps: Customer;
    isEdit: boolean;
}

interface OrdersPageFormData {
    email: string;
    name: string;
    lastName: string;
}

export const EditUserInfoForm: React.FC<EditUserInfoForm> = (props: EditUserInfoForm) => {
    const { register, handleSubmit, errors } = useForm<OrdersPageFormData>();

    const [userForm, setUserForm] = useState<Customer>(props.editProps);
    const auth = useContext(ClientContext);

    const updateInfoHandler = async (): Promise<void> => {
        if (props.isEdit) {
            const data: Data = await CustomerService.updateCustomer(JSON.stringify(userForm), {
                'Content-Type': 'application/json',
                Authorization: `Bearer ${auth.token}`,
            });
            toaster.notify(data.message, {
                duration: 2000,
            });
        } else {
            return;
        }
        props.closeModal();
    };

    useEffect(() => {
        if (props.isEdit) {
            setUserForm(props.editProps);
        } else {
            setUserForm({ email: '', lastName: '', name: '', order: [], password: '' });
        }
    }, [props.isEdit, props.editProps]);

    const userInfoChangeHandler = (event: InputEvent): void => {
        setUserForm({ ...userForm, [event.target.name]: event.target.value });
    };

    return (
        <Container
            onClick={(event: React.MouseEvent<HTMLDivElement>) => handleClickOutside(event, 'overlay-edit-user', props)}
            id={'overlay-edit-user'}
            fluid={true}
            className={props.show ? 'show-modal add-modal-wrapper' : 'hide-modal'}
        >
            <div key={1} className="d-flex justify-content-around align-items-center">
                <div className="change-user-info-form ">
                    <h3>Change User form</h3>
                    <label htmlFor="name">Your name</label>
                    <input
                        onChange={userInfoChangeHandler}
                        type="text"
                        className={'form-control'}
                        value={userForm.name}
                        name="name"
                        id="name"
                        placeholder="name"
                        ref={register({ required: true })}
                    />
                    <ErrorMessage error={errors.name} type={'error'} />
                    <label htmlFor="email">Your last name</label>
                    <input
                        onChange={userInfoChangeHandler}
                        type="text"
                        className={'form-control'}
                        value={userForm.lastName}
                        name="lastName"
                        id="lastName"
                        placeholder="lastName"
                        ref={register({ required: true })}
                    />
                    <ErrorMessage error={errors.lastName} type={'error'} />

                    <label htmlFor="email">Your email</label>
                    <input
                        onChange={userInfoChangeHandler}
                        type="text"
                        className={'form-control'}
                        value={userForm.email}
                        name="email"
                        id="email"
                        placeholder="email"
                        ref={register({ required: true, pattern: /^\S+@\S+$/i })}
                    />
                    <ErrorMessage error={errors.email} type={'error'} />
                    <button onClick={handleSubmit(updateInfoHandler)} className="button-book">
                        Update Info
                    </button>
                    <button className={'close-modal-button'} onClick={() => props.closeModal()}>
                       <Close/>
                    </button>
                </div>
            </div>
        </Container>
    );
};
