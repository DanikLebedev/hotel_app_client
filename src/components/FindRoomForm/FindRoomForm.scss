@import "../../styles/variables";

.fixed-form-wrapper {
  display: flex;
  align-items: center;
  position: fixed;
  top: 30%;
  right: 0;
  z-index: 15;
  transition: .6s ease-in-out;
  animation: shake 3s infinite;

  .toggle-fixed-form-button {
    border-radius: 3px;
    color: #ffffff;
    text-align: center;
    font-size: 1.5em;
    cursor: pointer;
    background: $price-color;
    padding: 10px;

  }

  .fixed-form {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
}

.fixed-form-active {
  right: 240px;
  animation: stopShake .1ms infinite;
}

@keyframes stopShake {
  from {
    transform: translateX(220px)
  } to {
  transform: translateX(220px)
      }
}

.find-room-form {
  border: 2px solid $price-color;
  padding: 20px 25px;
  background: #ffffff;
  border-radius: 3px;
  display: flex;
  justify-content: center;
  flex-direction: column;

  label {
    font-size: 1.2em;
    color: $price-color;
    font-weight: bold;
    margin: 5px;
  }

  select {
    border-radius: 3px;
    border: 1px solid $price-color;
    padding: 10px;
    margin: 10px;
    color: $price-color;
  }

  input {
    border-radius: 3px;
    border: 1px solid $price-color;
    padding: 10px;
    margin: 10px;
  }

  input[type=date] {
    -webkit-appearance: none;
    color: $price-color;

    &::placeholder {
      color: $price-color;
    }
  }

  input[type=date]::-webkit-inner-spin-button {
    display: none;
    -webkit-appearance: none;
  }

}

.find-room-button {
  color: #fff;
  background-color: $price-color;
  font-weight: 700;
  height: 50px;
  border: none;
  display: block;
  font-size: 1.2em;
  width: 100%;
  border-radius: 3px;
}


@keyframes shake {
  from {
    transform: translateX(220px);
  }
  50% {
    transform: translateX(210px);
  }

  to {
    transform: translateX(220px);
  }
}


@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 1) {

}

@media screen and (min-width: 768px) and (max-width: 1199px) {
  .fixed-form-wrapper {
    .toggle-fixed-form-button {
      font-size: 1rem;
    }
  }

  .find-room-button {
    font-size: 1rem;
  }
}

@media screen and (max-width: 768px) and (min-width: 480px) {
  .fixed-form-wrapper {
    .toggle-fixed-form-button {
      font-size: 1rem;
    }
  }

  .find-room-button {
    font-size: 1rem;
  }
}

@media screen and (min-width: 300px) and (max-width: 480px) {

  .fixed-form-wrapper {
    right: 10px;

    .toggle-fixed-form-button {
      font-size: 0.8em;
      width: 80px;
    }
  }

  .fixed-form-active {
    right: 240px;
  }

  .find-room-button {
    font-size: 0.8rem;
  }


  .find-room-form {
    font-size: 0.8em;
  }

}