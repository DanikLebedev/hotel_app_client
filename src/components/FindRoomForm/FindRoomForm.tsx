import React, { useState } from 'react';
import './FindRoomForm.scss';
import { Container } from 'react-bootstrap';

export const FindRoomForm = () => {
    const [showForm, setShowForm] = useState(false);

    const ToggleFormHandler = () => {
        setShowForm(!showForm);
    };
    return (
        <div className={showForm ? 'fixed-form-wrapper fixed-form-active' : 'fixed-form-wrapper'}>
            <div className="toggle-fixed-form" onClick={ToggleFormHandler}>
                Book Room
            </div>
            <div>
                <div className="find-room-form">
                    <label htmlFor={'checkIn'}>CheckIn</label>
                    <input type="date" id="checkIn" name="checkIn" />
                    <label htmlFor={'checkOut'}>CheckOut</label>
                    <input type="date" id="checkIn" name="checkOut" />
                    <select name="" id="">
                        <option value="">asd</option>
                        <option value="">asd</option>
                        <option value="">asd</option>
                        <option value="">asd</option>
                    </select>
                    <button>Check rooms</button>
                </div>
            </div>
        </div>
    );
};
