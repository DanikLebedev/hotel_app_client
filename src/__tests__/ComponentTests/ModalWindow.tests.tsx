import { mount, shallow } from 'enzyme';
import FindRoomForm from '../../components/FindRoomForm/FindRoomForm';
import React from 'react';
import { EditUserInfoForm } from '../../components/EditUserInfoForm/EditUserInfo';
it('should render without crashing', function() {
    const closeModal = jest.fn();
    const wrapper = shallow(
        <EditUserInfoForm
            closeModal={closeModal}
            editProps={{ email: '', password: '', name: '', lastName: '', order: [] }}
            isEdit={false}
            show={false}
        />,
    );
    expect(wrapper).toMatchSnapshot();
});
