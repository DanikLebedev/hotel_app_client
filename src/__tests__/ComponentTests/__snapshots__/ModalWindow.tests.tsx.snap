// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<Container
  className="hide-modal"
  fluid={true}
  id="overlay"
  onClick={[Function]}
>
  <div
    className="d-flex justify-content-around align-items-center"
    key="1"
  >
    <div
      className="change-user-info-form"
    >
      <h3>
        Change User form
      </h3>
      <label
        htmlFor="name"
      >
        Your name
      </label>
      <input
        className="form-control"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="name"
        type="text"
        value=""
      />
      <ErrorMessage
        type="error"
      />
      <label
        htmlFor="email"
      >
        Your last name
      </label>
      <input
        className="form-control"
        id="lastName"
        name="lastName"
        onChange={[Function]}
        placeholder="lastName"
        type="text"
        value=""
      />
      <ErrorMessage
        type="error"
      />
      <label
        htmlFor="email"
      >
        Your email
      </label>
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="text"
        value=""
      />
      <ErrorMessage
        type="error"
      />
      <button
        className="button-book"
        onClick={[Function]}
      >
        Update Info
      </button>
      <button
        className="close-modal-button"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f410",
                "M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z",
              ],
              "iconName": "window-close",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>
</Container>
`;
